File: ios/SmartyAlerts.m
#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>
#import <React/RCTBridgeMethod.h>

@interface RCT_EXTERN_MODULE(SmartyAlerts, NSObject)

RCT_EXTERN_METHOD(alert:(NSString)title
                  withMessage:(NSString)message
                  withBtnOneLabel:(NSString)btnOneLabel
                  withBtnTwoLabel:(NSString)btnTwoLabel
                  withResolve:(RCTResponseSenderBlock)resolve
                  withReject:(RCTResponseSenderBlock)reject)

@end

File: ios/SmartyAlerts.swift
import SCLAlertView

@objc(SmartyAlerts)
class SmartyAlerts: NSObject {
  
  @objc(alert:withMessage:withBtnOneLabel:withBtnTwoLabel:withResolve:withReject:)
  func alert(title: String,
             message: String,
             btnOneLabel: String,
             btnTwoLabel: String,
             resolve: @escaping RCTResponseSenderBlock,
             reject: @escaping RCTResponseSenderBlock) -> Void {
    DispatchQueue.main.async {
      let appearance = SCLAlertView.SCLAppearance(showCloseButton: false, contentViewCornerRadius: CGFloat(16), fieldCornerRadius: CGFloat(25), buttonCornerRadius: CGFloat(16))
      let alertView = SCLAlertView(appearance: appearance)
      alertView.addButton(btnOneLabel, backgroundColor: UIColor.purple, textColor: UIColor.white) {
        resolve([])
      }
      if(btnTwoLabel != "") {
        alertView.addButton(btnTwoLabel, backgroundColor: UIColor.purple, textColor: UIColor.white) {
          reject([])
        }
      }
      alertView.showNotice(title, subTitle: message)
    }
  }
}
